### PR TITLE
docs: best practices

### PR DESCRIPTION
**Pull request by aidenybai**
Adds new documentation + deopts #342 

Preview new docs: https://million-git-docs-best-practices-millionjs.vercel.app/

**Please describe the changes this PR makes and why it should be merged:**

**Status**

- [x] Code changes have been tested against prettier, or there are no code changes
- [x] I know how to update typings and have done so, or typings don't need updating

**Semantic versioning classification:**

- [x] This PR changes the codebase
  - [ ] This PR includes breaking changes (methods removed or renamed, parameters moved or removed)
  - [x] This PR changes the internal workings with no modifications to the external API (bug fixes, performance improvements)
- [ ] This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

**Comment by tobySolutions** [comment]
Thanks, Aiden!! I'm on it.

**Comment by tobySolutions** [comment]
I made a lot of updates to the "Thinking in Million.js" and "Rules of Blocks" pages @aidenybai. Can you please review this? Also, I need help with completing the "Unexpected behavior" section of the "Rules of blocks"

Thanks!!

**Comment by tobySolutions** [comment]
Good job @aidenybai! 🚀🚀🚀.

File: website/pages/docs/rules.mdx [hunk 1]
```diff
@@ -225,34 +225,55 @@ const MyComponentBlock = block(MyComponent) // ❌ ERROR
...
+      )
+    }
+    ```
 <br/>
 
 <Callout>You can only use block() with a function declaration or arrow function</Callout>
 
-   As the above says, you cannot use multiple return in blocks. This is unsupported at the moment. This is how you should return when you create a `block()`. A more subtle case:
+   Well, leagacy React has it such that we can have classes be used to create components, however that is old syntax now, but, if you happen to be finding it hard to be getting the error above; it might be because you
+   aren't using function-components. To resolve this, all you have to do is to __Use Functional Components__.
```
Comment (aidenybai) [inline website/pages/docs/rules.mdx:277]: the double __ here might cause a parsing error

Comment (tobySolutions) [reply website/pages/docs/rules.mdx]: I don't think that's the problem, cos, I just fixed it

**Comment by tobySolutions** [comment]
Awesome!! Almost done, I guess?

**Pull request merged**